### DISCUSSION TITLE
Problems with BCI package

### DISCUSSION THREAD

**Discussion by emmamclea** [IBAHCM/RPiR#484] [Archive] [answered]
Hi @richardreeve, @soniamitchell 
I have been working through my projects and haven't had any problems with the BCI package. When I *devtools::document()* on my project I now get the error: 
*Error in is.data.frame(x) : object 'bci_2010' not found* 
However, I can run through each line of code and create an HTML file with no problems. When I look at the traceback I get what is shown on the image. 
Do either of you have any idea what may be happening or how to resolve this? 

Thanks, 
Emma 
![2022-01-21 (2)](https://user-images.githubusercontent.com/94693682/150522999-59390e41-71b6-44f5-be90-30950a44ce09.png)

**Comment by emmamclea** [comment]
Update: I thought if I tried to delete and redownload the BCI package it may work however I am now getting the following error.
![2022-01-21 (1)](https://user-images.githubusercontent.com/94693682/150569399-81116431-8044-4646-8a46-9c7c87b37203.png)

**Comment by richardreeve** [comment, marked as answer]
@emmamclea - I don't think the problem was with the BCI package installation, but you do need to add it as a dependency, and the problem with your attempt to add it was that this isn't how you use `use_dev_package()`. You say `use_dev_package("emmamcleaBCI", remote="IBAHCM/emmamcleaBCI")`. As well as adding it as a dependency like this, you need to `@import` it in the package R file in the R folder like you do for ggplot2.

It's possible that the original problem with using `bci_2010` is that in "tree species populations.R" (in the R folder?) is that you refer to `bci_2010` directly in that file. Because it's in another package, you should refer to it as `emmamcleaBCI::bci_2010`.

**Reply by emmamclea** [reply]
Ah, I think I was going wrong by forgetting the "remote=" section thank you for that! Not sure why it happened in the first place though. 
I have a quick follow-up question if that's okay? When I am using demo(emmamcleaDiversity) I get the names of my demos but not the description I have added to them. Is there a certain way to make this happen? I have tried many different spaces between the name of the demo and the description but I can't seem to get it to work.

**Reply by richardreeve** [reply]
The most common reason for the `demo()` function not working is that you have mistyped the exact name of the demo file (or you've forgotten to take off the `.R`) in the `demo/00Index` file. This is almost always the first problem. Any misspelling and you lose all of the descriptions. Secondly, you need at least 3 spaces. The only other thing is misspelling `00Index` itself. That's it basically. Hope that works for you!

**Reply by emmamclea** [reply]
Hi Richard, 
Happy to say this worked! I had not capitalized 'I' in Index, however, I was getting the names of the demos but not the description. It was all very confusing but thank you for the help! :)
